wagtail/admin/views/bulk_actions/move.py

```diff
@@ -69,35 +72,39 @@ def move_confirm(request, parent_page_id, dest_page_id):
 
     page_ids = list(map(int, request.GET.getlist('id')))
     pages_to_move = []
+    pages_with_no_access = []
     for _page in get_list_or_404(Page, id__in=page_ids):
         page_to_move = _page.specific
         page_perms = page_to_move.permissions_for_user(request.user)
         if not page_perms.can_move():
-            raise PermissionDenied
+            pages_with_no_access.append(page_to_move)
+            continue
         if not page_perms.can_move_to(destination):
-            raise PermissionDenied
+            pages_with_no_access.append(page_to_move)
+            continue
         if not Page._slug_is_available(page_to_move.slug, destination, page=page_to_move):
             messages.error(
                 request,
                 _("The slug '{0}' is already in use at the selected parent page. Make sure the slug is unique and try again").format(page_to_move.slug)
             )
             return redirect(reverse('wagtailadmin_bulk_move', args=[parent_page_id, dest_page_id]) + '?' + urlencode([('id', page_id) for page_id in page_ids]))
-        for fn in hooks.get_hooks('before_move_page'):
-            result = fn(request, page_to_move, destination)
-            if hasattr(result, 'status_code'):
-                return result
         pages_to_move.append(page_to_move)
 
     if request.method == 'GET':
         return TemplateResponse(request, 'wagtailadmin/pages/bulk_actions/confirm_bulk_move.html', {
             'num_pages': len(page_ids),
             'pages_to_move': pages_to_move,
+            'pages_with_no_access': pages_with_no_access,
             'destination': destination,
             'submit_url': reverse('wagtailadmin_bulk_move_confirm', args=[parent_page_id, dest_page_id])
             + '?' + urlencode([('id', page_id) for page_id in page_ids]),
         })
     else:
         for page_to_move in pages_to_move:
+            for fn in hooks.get_hooks('before_move_page'):
+                result = fn(request, page_to_move, destination)
+                if hasattr(result, 'status_code'):
+                    return result
```

Comment:
@KIRA009 can you add a note to revisit this? We discussed we'd cover this during the next core meeting.
I am not convinced this should apply for bulk actions but we're split on this so wider consensus is needed